csharp/src/ProtocolBuffers.Test/GeneratedMessageTest.cs

```diff
@@ -147,6 +158,65 @@ public void RoundTrip_RepeatedValues()
         }
 
         [Test]
+        public void RoundTrip_Maps()

```

Comment:
May want more tests for the generated message. This change has enough tests for MapField, should add tests for the generated MapEntry too.